Cpp_test/lib/ipc-shm-receive.cpp

```diff
@@ -0,0 +1,149 @@
+#include "ipc-shm-receive.h"
+
+IPCShmReceive::~IPCShmReceive()
+{
+
+	if (this->shm_ptr != nullptr)
+	{
+		if (this->shm_ptr->checkLockMutex == true)
+		{
+			this->shm_ptr->checkLockMutex = false;
+		}
+		int ret1 = pthread_cond_destroy(&(this->shm_ptr->cond));
+		if (ret1 != 0)
+		{
+			std::cerr << "Error at pthread_cond_destroy(): " << strerror(ret1) << std::endl;
+		}
+
+		int ret2 = pthread_mutex_trylock(&(this->shm_ptr->mutex));
+		if (ret2 != 0)
+		{
+			std::cerr << "Error at pthread_mutex_trylock(): " << strerror(ret2) << std::endl;
```

Comment:
If the sender has destroyed the mutex you will have an error there "invalid argument" but that's normal. Maybe you should check this case of error and not print anything because it's normal.